src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,175 @@
+import Html.Styled.Attributes as Attrs exposing (css)
+import TabsInternal.V2 as TabsInternal
+
+
+{-| Builds a carousel with previous and next controls
+Returns:
+`controls`: previous and next buttons element
+`slides` the container with the carousel contents
+`containerAttributes` attributes that should be used on the parent div of both the button and slides elements
+-}
+viewWithPreviousAndNextControls :
+    { selected : id
+    , panels :
+        List
+            { id : id
+            , slideHtml : Html msg
+            , ariaLabel : AriaLabel
+            }
+    , viewPreviousButton : Html msg
+    , viewNextButton : Html msg
+    , ariaLabel : AriaLabel
+    , controlListStyles : List Style
+    }
+    ->
+        { controls : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithPreviousAndNextControls config =
+    { controls =
+        div [ css config.controlListStyles ]
+            [ config.viewPreviousButton, config.viewNextButton ]
+    , slides =
+        List.map
+            (\panel ->
+                Html.div
+                    [ Attrs.attribute "role" "group"
+                    , Aria.roleDescription "slide"
+                    , ariaLabelToAttr panel.ariaLabel
+                    , css
+                        [ if config.selected == panel.id then
+                            Css.display Css.block
+
+                          else
+                            Css.display Css.none
+                        ]
+                    ]
+                    [ panel.slideHtml ]
+            )
+            config.panels
+            |> Html.div [ Attrs.attribute "atomic" "false", Attrs.attribute "live" "polite" ]
+    , containerAttributes =
+        [ Attrs.attribute "role" "region"
+        , Aria.roleDescription "carousel"
+        , ariaLabelToAttr config.ariaLabel
+        ]
+    }
+
+
+ariaLabelToAttr : AriaLabel -> Attribute msg
+ariaLabelToAttr label =
+    case label of
+        IdLabel l ->
+            Aria.labeledBy l
+
+        StringLabel l ->
+            Aria.label l
+
+
+{-| Type which represents the type of aria label which will be used
+`IdLabel` will point to an existing element id on the DOM
+`StringLabel` will be a label of the element
+-}
+type AriaLabel
+    = IdLabel String
+    | StringLabel String
+
+
+{-| Builds a carousel with tab buttons
+Returns:
+`controls`: tabs control buttons
+`slides` container with the carousel contents
+-}
+viewWithTabControls :
+    { cfg
+        | selected : id
+        , panels :
+            List
+                { id : id
+                , slideHtml : Html msg
+                , tabControlHtml : Html Never
+                , idString : String
+                }
+        , focusAndSelect : { select : id, focus : Maybe String } -> msg
+        , tabControlStyles : Bool -> List Style
+        , tabControlListStyles : List Style
+    }
+    ->
+        { controls : Html msg
```

Comment:
I think we still need a top level container with appropriate attributes as in the function above!

Reply:
I didn't find any references to this container on the [docs](https://www.w3.org/WAI/ARIA/apg/patterns/carousel/) actually, do we really need it here?

Reply:
Soooo the description for Tabbed carousels says "The structure of a tabbed carousel is the same as a basic carousel except that: [...]".

Since the basic one has that container and there is no explicit instruction that we shouldn't have it in the tabbed version, I assumed it was reasonable to assume that we should keep it in the tabbed version.

Reply:
Ohhh, I totally ignored the "The structure of a tabbed carousel is the same as a basic carousel except that: [...]".
thing and my eyes moved directly to the bullet points haha